modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/SourceConnectionManager.java

```diff
@@ -49,11 +52,17 @@
    * @throws ConnectionException if a connection could not be obtained
    */
   <T> T getConnection(ConfigurationInstance config) throws ConnectionException {
-    ConnectionHandler<Object> connectionHandler = connectionManager.getConnection(config.getValue());
-    Object connection = connectionHandler.getConnection();
-    connections.put(new Reference<>(connection), connectionHandler);
-
-    return (T) connection;
+    lock.lock();
```

Comment:
using AtomicInteger makes the need for this lock unnecessary. Also for future reference, notice that the only critical resource here is the map. So if you were to actually synchronize, you should only synchronoize that part. It doesnt make sense to syncrhonize the connect part